### ISSUE TITLE
Enhance code comments

### ISSUE THREAD

**Issue by atsju** [ObKo/stm32-cmake#191]
Who has never read his own code and didn't understand it a year after ?

This repo becomes bigger, greater and more used every day and even if I'm able to write some cmake code I do not consider myself as expert. It is often quite difficult for me to understand exactly what the code does. 

I think a minimum of comment should be added and every new PR should have (in an ideal world) :

- comments for the added code
- documentation update if behavior changes

Also the original code should be commented in the futur.

Comment shall not explain what `set` does on every line but regex, general flow and "advanced" cmake features (I know this means nothing depending on people) could definitely be.

(#190 mentions this problem).

![image](https://user-images.githubusercontent.com/4628382/115136296-334c4980-a01f-11eb-9964-330c319f774d.png)

**Comment by gudnimg** [comment]
I agree. Its hard to see what the code is doing sometimes :sweat_smile:
